### DISCUSSION TITLE
Midi Functionality?

### DISCUSSION THREAD

**Discussion by ThePerezenator** [kyleneideck/BackgroundMusic#414] [Q&A]
Does this support midi? (moving faders on a midi controller changes volume of applications) I have this app called midi mixer on my windows bootcamp and would love to have this same sort of functionality when I'm in macos.

**Comment by kyleneideck** [comment]
It doesn't support MIDI, but that's a cool idea. If anyone would be interested in working on a PR for it, I'd be happy to help out.

**Reply by ThePerezenator** [reply]
Thanks for the response, hopefully it will be implemented soon.

**Reply by aledefra** [reply]
> It doesn't support MIDI, but that's a cool idea. If anyone would be interested in working on a PR for it, I'd be happy to help out.

Some time later, but I would love this idea too and I could also help in implementing this feature. Can I help with this?

**Reply by kyleneideck** [reply]
@aledefra Sure, that would be great. The easiest way is probably to fork this repo and start a work-in-progress branch that you can share. But if you find it straightforward enough, feel free to just make a PR. You can also make a draft PR if that helps.

It might also be worth discussing the design up front a bit. Do you have an idea of how it would fit into the UI? I haven't really thought about it, but I was imagining it as just a window you can open from the Preferences menu where you select which MIDI controls map to which Background Music functions (volume sliders, mute buttons, etc.)

I haven't really looked into this either, but I'm also wondering if there's any existing open-source software that might make this easier. I'm sure there would be some libraries that are worth looking into at least. If we could find a project that listens for MIDI events and generates Apple Events, maybe the most pragmatic approach would be to extend Background Music's AppleScript interface and configure the MIDI project to use it. What do you think?
